### PR TITLE
[AUTO_IMPORT_NAMES] Failing fixture for #4650 to skip already in use statement

### PR DESCRIPTION
**Pull request by samsonasik**

### CODE REVIEW

**Comment by samsonasik** [comment]
fixed.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: phpstan.neon [hunk 1]
```diff
@@ -619,6 +619,7 @@ parameters:
         - '#Parameter \#1 \$node of method Rector\\DeadCode\\Rector\\Plus\\RemoveDeadZeroAndOneOperationRector\:\:refactor\(\) expects PhpParser\\Node\\Expr\\AssignOp\\Div\|PhpParser\\Node\\Expr\\AssignOp\\Minus\|PhpParser\\Node\\Expr\\AssignOp\\Mul\|PhpParser\\Node\\Expr\\AssignOp\\Plus\|PhpParser\\Node\\Expr\\BinaryOp\\Div\|PhpParser\\Node\\Expr\\BinaryOp\\Minus\|PhpParser\\Node\\Expr\\BinaryOp\\Mul\|PhpParser\\Node\\Expr\\BinaryOp\\Plus, PhpParser\\Node\\Expr\\AssignOp\|PhpParser\\Node\\Expr\\BinaryOp given#'
 
         - '#Cognitive complexity for "Rector\\CodeQuality\\Naming\\MethodCallToVariableNameResolver\:\:resolveVariableName\(\)" is 12, keep it under 9#'
+        - '#Cognitive complexity for "Rector\\CodingStyle\\Node\\NameImporter\:\:isFoundInUse\(\)" is 16, keep it under 9#'
```
Comment (TomasVotruba) [inline phpstan.neon:622]: This should be excluded a and method rewritten in less complexity.

File: rules/coding-style/src/Node/NameImporter.php [hunk 2]
```diff
@@ -88,6 +90,10 @@ public function importName(Name $name): ?Name
             return null;
         }
 
+        if ($this->isShortNameInUseStatement($name)) {
```
Comment (TomasVotruba) [inline rules/coding-style/src/Node/NameImporter.php:93]: This should be added to ClassNameImportSkipper class, as one of skippers

File: rules/coding-style/src/Node/NameImporter.php [hunk 3]
```diff
@@ -215,6 +231,35 @@ private function isNonExistingClassLikeAndFunctionFullyQualifiedName(Name $name)
         return ! function_exists($classOrFunctionName);
     }
 
+    private function isFoundInUse(Name $name): bool
+    {
+        $parent = $name->getAttribute(AttributeKey::PARENT_NODE);
+
+        while ($parent) {
```
Comment (TomasVotruba) [inline rules/coding-style/src/Node/NameImporter.php:238]: Try getAttribute(AttributeKey::USES) or similar to get those

File: phpstan.neon (hunk 1 above)
Comment (samsonasik) [reply phpstan.neon]: implemented.

File: rules/coding-style/src/Node/NameImporter.php (hunk 2 above)
Comment (samsonasik) [reply rules/coding-style/src/Node/NameImporter.php]: implemented.

File: rules/coding-style/src/Node/NameImporter.php (hunk 3 above)
Comment (samsonasik) [reply rules/coding-style/src/Node/NameImporter.php]: implemented.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba I think it is ready.

File: rules/coding-style/src/ClassNameImport/ClassNameImportSkipper.php [hunk 4]
```diff
@@ -35,4 +38,29 @@ public function shouldSkipNameForFullyQualifiedObjectType(
...
+            $useUses = $use->uses;
+            foreach ($useUses as $useUse) {
+                if ($useUse->name->getLast() === $name->getLast()) {
+                    return true;
+                }
+            }
+        }
+
+        return false;
+    }
```
Comment (TomasVotruba) [inline rules/coding-style/src/ClassNameImport/ClassNameImportSkipper.php]: This should be decoupled in own service that implements `ClassNameImportSkipVoterInterface`

Comment (TomasVotruba) [reply rules/coding-style/src/ClassNameImport/ClassNameImportSkipper.php]: Ah, this might require more refactoring. Skip it for now

**Comment by TomasVotruba** [comment]
Looks good, thank you :+1:

**Pull request merged**